packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutWithAddCardPanel/DashboardGridLayoutWithAddCardPanel.tsx

```diff
@@ -47,6 +47,40 @@ export class DashboardGridLayoutWithAddCardPanel extends BaseComponent<
     };
   }
 
+  public componentDidMount(): void {
+    if (this._cardsForAddCardPanel !== this.props.addCardPanelCards || this._cardsForLayout !== this.props.dashboardCards) {
+      this._cardsForAddCardPanel = this.props.addCardPanelCards;
+      this._cardsForLayout = this.props.dashboardCards;
+      const cardIds: string[] = [];
+      const cardNodes: JSX.Element[] = [];
+      const layout: DashboardGridBreakpointLayouts = { lg: [{ i: 'section0', y: 0, x: 0, size: CardSize.section }] };
+      layout.lg = layout.lg!.concat(this.props.layout.lg!);
+      this.props.dashboardCards.map((card: IDGLCard) => {
+        cardIds.push(card.id);
+        cardNodes.push(
+          <div key={card.id} id={card.id + 'dglCard'}>
+            {card.renderElement}
+          </div>
+        );
+      });
+      this.props.addCardPanelCards.map((card: IDGLCard) => {
```

Comment:
Same as above, use `forEach` if you are not using `.map` and the array it allocates.

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map